src/artemis/document.cljc

```diff
@@ -86,17 +86,23 @@
   (if (empty? fragments)
     sel-set
     (reduce (fn [acc sel]
-              (if (= (:node-type sel) :fragment-spread)
+              (cond
+                (= (:node-type sel) :fragment-spread)
                 (->> (get-in fragments [(:name sel) :selection-set] [])
                      (resolve-fragments fragments)
                      (into acc))
+
+                (not (nil? (:selection-set sel)))
+                (conj acc (update sel :selection-set #(resolve-fragments fragments %)))
+
+                :else
                 (conj acc sel)))
             []
             sel-set)))
 
 (defn- update-definitions [fragments definitions]
-  (map #(assoc % :selection-set (resolve-fragments fragments (:selection-set %)))
-       definitions))
+  (mapv #(assoc % :selection-set (resolve-fragments fragments (:selection-set %)))
```

Comment:
Inline fragments converted the selection set from a vector to a list, which caused an issue.

Reply:
slightly prettier
`#(update % :selection-set (partial resolve-fragments fragments))`